diff --git a/src/utils.js b/src/utils.js
index a9b2ac69c..6494b0081 100644
--- a/src/utils.js
+++ b/src/utils.js
@@ -41,7 +41,8 @@ function getLastOfPath(object, path, Empty) {
 
     const key = cleanKey(stack.shift());
     if (!object[key] && Empty) object[key] = new Empty();
-    object = object[key];
+    // prevent prototype pollution
+    if (object.hasOwnProperty(key)) object = object[key];
   }
 
   if (canNotTraverseDeeper()) return {};
