diff --git a/test/test.js b/test/test.js
index c88e0fda96..822d229e07 100644
--- a/test/test.js
+++ b/test/test.js
@@ -7554,6 +7554,17 @@
         skipAssert(assert);
       }
     });
+
+    QUnit.test('should not merge `Object.prototype` properties', function(assert) {
+      assert.expect(1);
+
+      _.merge({}, { 'constructor': { 'prototype': { 'a': 1 } } });
+
+      var actual = 'a' in objectProto;
+      delete objectProto.a;
+
+      assert.notOk(actual);
+    });
   }());
 
   /*--------------------------------------------------------------------------*/
