diff --git a/test/ducktype.test.js b/test/ducktype.test.js
index 0da68f6..d9c119d 100644
--- a/test/ducktype.test.js
+++ b/test/ducktype.test.js
@@ -878,6 +878,7 @@ test('url', function (t) {
   t.is(type.test('http:/example.com'), false);
   t.is(type.test('http//example.com'), false);
   t.is(type.test('http://example'), false);
+  t.is(type.test('http://example.com?a=2&b=a%2Bb'), true);
   t.is(type.test('www.google.com'), false);
 });
 
