diff --git a/test/index.js b/test/index.js
index f854c42..f90affc 100644
--- a/test/index.js
+++ b/test/index.js
@@ -42,6 +42,14 @@ describe('#set', function() {
     obj.a[1].should.equal('B');
     obj.a[2].should.equal('C');
   });
+
+  context('attempt to modify prototype', function() {
+    it('throws an error', function() {
+      assert.throws(
+        () => set({}, '__proto__.foo', 'hacker_attack!'),
+        Error, 'Cannot assign reference to prototype')
+    })
+  })
 });
 
 describe('#decorate', function() {
@@ -67,4 +75,4 @@ describe('#undecorate', function() {
     undecorate(obj);
     expect(obj.setKey).to.not.be.a('function');
   });
-});
\ No newline at end of file
+});
