diff --git a/mixin.js b/mixin.js
index e553cd7..68b44ff 100644
--- a/mixin.js
+++ b/mixin.js
@@ -158,7 +158,6 @@ function mixin(base, mixin) {
 
 function mixin_constructor(name, ctor) {
   // validate base name to avoid evil code injection
-  if(!/^[$A-Z_][0-9A-Z_$]*$/i.test(name)) return;
 
   var str = "function __ctor() { var c = ctor.constructors; for (var i in c) { c[i].apply(this, arguments); } };".replace(/__ctor/g, name);
   eval(str);