diff --git a/index.js b/index.js
index fab9efb..89f3a06 100755
--- a/index.js
+++ b/index.js
@@ -492,7 +492,7 @@ class NodeClam {
         try {
             await fs_access(path, fs.constants.R_OK);
             version_cmds_exec = version_cmds[scanner].split(' ');
-            const {stdout} = await cp_execfile(version_cmds_exec[0], [version_cmds_exec[2]]);
+            const {stdout} = await cp_execfile(version_cmds_exec[0], [version_cmds_exec[1]]);
             if (stdout.toString().match(/ClamAV/) === null) {
                 if (this.settings.debug_mode) console.log(`${this.debug_label}: Could not verify the ${scanner} binary.`);
                 return false;
